Tests/JaegerTests/JaegerTracerTests.swift

```diff
@@ -10,107 +10,107 @@ import XCTest
 import CoreData
 
 class JaegerTracerTests: XCTestCase {
-    
+
     private var coreDataStack: CoreDataStack!
-    
+
     override func setUp() {
         coreDataStack = newCDStack()
     }
-    
+
     private func newCDStack() -> CoreDataStack {
         let stack = CoreDataStack(
             modelName: TestUtilities.Constants.coreDataAgentModelName,
             model: TestUtilities.modelForCoreDataAgent,
             type: .inMemory
         )
-        
+
         return stack
     }
-    
+
     func testCreateSpan() {
         // Setup
         let reachability = TestReachabilityTracker(reachability: true)
         let sender = TestSender { _ in }
-        
+
         guard let CDAgentConfig = CDAgentConfiguration(
             averageMaximumSpansPerSecond: 1,
             savingInterval: 1,
             sendingInterval: 2,
             coreDataFolderURL: nil
             ) else {
-                return XCTFail()
+                return XCTFail("Invalid CDAgentConfig")
         }
-        
+
         let agent = CDAgent<JaegerSpan>(
             config: CDAgentConfig,
             sender: sender,
             stack: coreDataStack,
             reachabilityTracker: reachability
         )
-        
+
         // TEST
         let tracer = JaegerTracer(agent: agent)
         let otSpan = tracer.startRootSpan(operationName: "TESTSPANNAME")
-        var span: Span? = nil
+        var span: Span?
         let getSpanExpectation = XCTestExpectation(description: "get Span")
-        
+
         otSpan.get { getSpan in
             span = getSpan
             getSpanExpectation.fulfill()
         }
-        
+
         wait(for: [getSpanExpectation], timeout: 0.5)
-        
+
         guard let finalSpan = span else {
-            return XCTFail()
+            return XCTFail("Get span should always return the underlying span.")
         }
-        
+
         XCTAssertEqual(finalSpan.operationName, "TESTSPANNAME")
         print(finalSpan.startTime.timeIntervalSince(Date()))
         XCTAssertLessThan(Date().timeIntervalSince(finalSpan.startTime), 0.5)
         XCTAssertEqual(finalSpan.isCompleted, false)
         XCTAssertEqual(finalSpan.flag, .sampled)
         XCTAssertEqual(finalSpan.spanRef.traceId, tracer.tracerId)
     }
-    
+
     func testReportSpan() {
-        
+
         // Setup
         let reachability = TestReachabilityTracker(reachability: true)
-        
+
         let spansSent = XCTestExpectation(description: "Agent sent spans")
         var sentSpans: [JaegerSpan] = []
         let sender = TestSender { spans in
-            
+
             guard let sentJaegerSpans = spans as? [JaegerSpan] else {
-                XCTFail()
+                XCTFail("Need JaegerSpans!")
                 spansSent.fulfill()
```

Comment:
Why is this side-effect after an `XCTFail`?

Reply:
I want to end the test after failing and not wait for the `XCTestExpectation` to also fail after 1 seconds: `wait(for: [spansSent], timeout: 1)`.

Reply:
Gotcha, let‘s just add a comment after the `fufill` that the purpose is to skip waiting for the fulfillment timeout.

Reply:
done :)